processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is the detector type string
+	TypeStr = "system"
+)
+
+var _ internal.Detector = (*Detector)(nil)
+
+// Detector is a system metadata detector
+type Detector struct {
+	provider systemMetadata
+}
+
+// NewDetector creates a new system metadata detector
+func NewDetector() (internal.Detector, error) {
+	return &Detector{provider: &systemMetadataImpl{}}, nil
+}
+
+// Detect detects system metadata and returns a resource with the available ones
+func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	res.InitEmpty()
+	attrs := res.Attributes()
+
+	hostname, err := d.provider.Hostname()
+	if err != nil {
+		return res, err
+	}
+
+	hostType, err := d.provider.HostType()
+	if err != nil {
+		return res, err
+	}
+
+	if d.provider.FQDNAvailable() {
+		fqdn, err := d.provider.FQDN(ctx)
+		if err != nil {
+			return res, err
+		}
+
+		attrs.InsertString(conventions.AttributeHostName, fqdn)
+	}
+
+	attrs.InsertString(conventions.AttributeHostHostname, hostname)
```

Comment:
host.hostname was removed from spec. I filed issue just now to remove throughout (https://github.com/open-telemetry/opentelemetry-collector/issues/2029). Don't think I would add it here now.

If we need to collect both hostname and the fqdn we may need to propose adding a `host.fqdn` semantic convention. Or if we could somehow make the detectors configurable and fqdn could be an option that might work too. But I think that would require refactoring the config for detectors.

Reply:
Oh, I didn't realize, thanks for pointing it out. An alternative that I considered for this was `net.host.name` but I am not sure if it would be really equivalent.

Reply:
Looks like net.host.name is part of [trace semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/9a803247d8fb652f9cf7b11270820415ed5dd2ce/specification/trace/semantic_conventions/span-general.md) though, not resource conventions so won't work here.

> Otherwise, maybe it would be most useful to use the FQDN as host.name?

Sounds good.